File: problems/TypeScript/004_is_palindrome/Test/is_palindrome_tests.ts
import * as path from "path";
async function load(fp: string) { return await import(path.resolve(fp.replace(/\.ts$/,""))); }
async function runTests(fp: string) {
    const m = await load(fp); const f = m.isPalindrome;
    const cases: [string,boolean,string][] = [["racecar",true,"racecar"],["hello",false,"hello"],["madam",true,"madam"],["level",true,"level"],["python",false,"python"],["a",true,"single"],["",true,"empty"]];
    let passed=0,failed=0; const results: any[]=[];
    for(const [s,exp,label] of cases){try{const r=f(s);if(r===exp){passed++;results.push({test:label,status:"pass"})}else{failed++;results.push({test:label,status:"fail",expected:exp,got:r})}}catch(e:any){failed++;results.push({test:label,status:"error",message:e.message})}}
    return {passed,failed,total:cases.length,results};
}
const fp=process.argv[2]??"starter.ts"; runTests(fp).then(o=>console.log(JSON.stringify(o,null,2)));

File: problems/Python/008_is_palindrome/Test/is_palindrome_tests.py
import importlib.util, sys

def load(fp):
    spec = importlib.util.spec_from_file_location("s", fp)
    m = importlib.util.module_from_spec(spec); spec.loader.exec_module(m); return m

def run_tests(fp):
    f = load(fp).is_palindrome
    cases = [("racecar",True,"racecar"),("hello",False,"hello"),("madam",True,"madam"),("A man a plan a canal Panama",True,"with spaces"),("python",False,"python"),("level",True,"level"),("",True,"empty"),("a",True,"single char")]
    passed=failed=0; results=[]
    for s,exp,label in cases:
        try:
            r=f(s)
            if r==exp: passed+=1; results.append({"test":label,"status":"pass"})
            else: failed+=1; results.append({"test":label,"status":"fail","expected":exp,"got":r})
        except Exception as e: failed+=1; results.append({"test":label,"status":"error","message":str(e)})
    return {"passed":passed,"failed":failed,"total":len(cases),"results":results}

if __name__=="__main__":
    print(run_tests(sys.argv[1] if len(sys.argv)>1 else "starter.py"))
